File: 03-PerfectlySphericalHousesInAVacuum/delivery.py
def lector(rute):
    with open(rute, 'r') as file:
        directions = file.read()
        #print("len: ", directions)
        santa_x, santa_y = 0, 0
        robo_x, robo_y = 0, 0
        visited_houses = {(0, 0)}  # Starting house

        # Use a toggle variable to switch between Santa and Robo-Santa
        is_santa_turn = True

        for direction in directions:
            # Determine which Santa is moving
            current_x, current_y = (santa_x, santa_y) if is_santa_turn else (robo_x, robo_y)

            if direction == '^':
                current_y += 1
            elif direction == 'v':
                current_y -= 1
            elif direction == '>':
                current_x += 1
            elif direction == '<':
                current_x -= 1

            visited_houses.add((current_x, current_y))

            # Update the corresponding Santa's position
            if is_santa_turn:
                santa_x, santa_y = current_x, current_y
            else:
                robo_x, robo_y = current_x, current_y

            # Toggle the turn for the next move
            is_santa_turn = not is_santa_turn
        print("houses visited: ", len(visited_houses))
        

rute_file = 'input.txt'
lector(rute_file)

File: 03-PerfectlySphericalHousesInAVacuum/delivery.js
const fs = require('fs');

function lector(rute) {
    const directions = fs.readFileSync(rute, 'utf-8');
    let santa_x = 0, santa_y = 0;
    let robo_x = 0, robo_y = 0;
    const visitedHouses = new Set(['0,0']);  // Starting house

    // Use a toggle variable to switch between Santa and Robo-Santa
    let isSantaTurn = true;

    for (const direction of directions) {
        // Determine which Santa is moving
        let currentX, currentY;
        if (isSantaTurn) {
            currentX = santa_x;
            currentY = santa_y;
        } else {
            currentX = robo_x;
            currentY = robo_y;
        }

        switch (direction) {
            case '^':
                currentY++;
                break;
            case 'v':
                currentY--;
                break;
            case '>':
                currentX++;
                break;
            case '<':
                currentX--;
                break;
        }

        visitedHouses.add(`${currentX},${currentY}`);

        // Update the corresponding Santa's position
        if (isSantaTurn) {
            santa_x = currentX;
            santa_y = currentY;
        } else {
            robo_x = currentX;
            robo_y = currentY;
        }

        // Toggle the turn for the next move
        isSantaTurn = !isSantaTurn;
    }

    console.log("Houses visited:", visitedHouses.size);
}

const ruteFile = 'input.txt';
lector(ruteFile);
